client/src/sim.rs

```diff
@@ -75,6 +80,10 @@ impl Sim {
         self.instantaneous_velocity = v;
     }
 
+    pub fn toggle_no_clip(&mut self) {
+        self.toggle_no_clip = true;
```

Comment:
Why not toggle `no_clip` directly here, as the name suggests?

Reply:
If `no_clip` is toggled immediately until waiting after the next input was sent, there would be a "retcon" for the frames after the last tick, which would cause a jolt if no-clip is toggled while moving. The simplest fix for anything like this is to delay the toggling until the next tick. The main downside is input lag, but this doesn't matter for non-canon things like no-clip toggling.

It might be easier to read this if there's a field called something like `queued_no_clip` or `no_clip_next_tick`, so the setting rather than the changes to be made to a setting are queued up.

Reply:
That makes sense. Please always justify non-obvious behavior like this in comments at minimum.

Reply:
Will do. Sometimes, it's a bit unclear to me what's obvious and what isn't, but hopefully, I can get better at this with experience.